consensus/spos/bls/subroundEndRound.go

```diff
@@ -87,6 +87,12 @@ func (sr *subroundEndRound) doEndRoundJob() bool {
 	sr.Header.SetPubKeysBitmap(bitmap)
 	sr.Header.SetSignature(sig)
 
+	err = sr.BlockProcessor().ApplyValidatorStatistics(sr.Header)
```

Comment:
consensus should stay as clean as possible. Consensus operate in state changes, does not care what the underlying implementation is.

Reply:
the problem with this is that consensus signs a block, and the same block is changes before broadcasting and sealing it with leaders signature. In the end, when we have to decide who was the malicious, we might think all those who signed the block, but it was only the leader who altered the final state. But the signature of the validators is on it.